src/gui/tutorial.cpp

```diff
@@ -696,6 +715,18 @@ void FurnaceGUI::drawTutorial() {
       (canvasW-ImGui::GetWindowSize().x)*0.5,
       (canvasH-ImGui::GetWindowSize().y)*0.5
     ));
+
+    if (tutorial.popupTimer<40) {
```

Comment:
I am not sure whether this is ideal. It doesn't look good to me.
Having "copied!" appear in a corner is something the user may not notice while looking at a popup in the center of the screen.

Reply:
the "copied!" popup appears at the bottom center of the main popup, not at a corner

Reply:
Got it, but still a bit hard to notice.
If the welcome window is too tall, it won't fit.

Also, consider changing the timer to a double and decreasing it by DeltaTime. This will prevent discrepancies on different refresh rates.

Reply:
when is it going to be too tall?

Reply:
i should note that "copied!" appears at the bottom of the window, not at the bottom of the content (if that makes sense).
its like `position: absolute; bottom: 10;` in css terms.
so even if the content doesnt fit all, the popup will appear

Reply:
> when is it going to be too tall?

On small screens.

Reply:
> > when is it going to be too tall?
> 
> On small screens.

how small do you want me to go?
![image](https://github.com/user-attachments/assets/53afd3a4-2414-4b07-9b34-b5cba6d3a968)
oh and mobile? no copying for mobile.

Reply:
Don't worry about it.

Reply:
> On mobile you could check for long hold (see the macro in gui.h).

I can't. The moment it's pressed it opens the browser.

Reply:
how?
i don't think there's an ImGui function for that

Reply:
Adding button behavior to text.
I'll continue the rest - don't worry.